apps/web/lib/api/links/process-link.ts

```diff
@@ -261,9 +302,7 @@ export async function processLink({
       // make sure projectId is set to the current workspace
       projectId: workspace?.id || null,
       // if userId is passed, set it (we don't change the userId if it's already set, e.g. when editing a link)
-      ...(userId && {
-        userId,
-      }),
+      userId: userId || payload.userId || null,
```

Comment:
why is this change required?

Reply:
Actually I think we can just make `userId` optional on `ProcessedLinkProps` instead ([commit](https://github.com/dubinc/dub/pull/631/commits/e479f14bfa02be6adafc1e497613b15bb5d32dea))